src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Unix.cs

```diff
@@ -26,69 +26,79 @@ public static Architecture OSArchitecture
         {
             get
             {
-                lock (s_osLock)
+                Debug.Assert(sizeof(Architecture) == sizeof(int));
+
+                if (s_osArch == -1)
                 {
-                    if (null == s_osArch)
+                    lock (s_osLock)
```

Comment:
Why lock at all?  Seems like these could just use optimistic concurrency, and if there's a race condition where on first accesses multiple threads are racing to initialize it, well, they'll do so and just all end up setting it to the same value.

Reply:
> Why lock at all? 

We are very close to the release and I am not very familiar with this particular code so I just wanted to minimize the risk assuming that there most probably is a reason why somebody added locks in the past ;)

Reply:
There is no reason this needs to lock.  The implementation in the shim is literally just returning a constant:
https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/Native/Unix/System.Native/pal_runtimeinformation.c#L45-L58
https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/Native/Unix/System.Native/pal_runtimeinformation.c#L65-L78
If we're going to fix the code, we should fix it right.

Reply:
> If we're going to fix the code, we should fix it right.

@stephentoub you are 100% right. I've applied your suggestions, PTAL